warehouse/static/js/warehouse/controllers/notification_controller.js

```diff
@@ -41,7 +41,7 @@ export default class extends Controller {
     // Check if the target is dismissable, and if so:
     // * whether it has no notificationId (it's ephemeral)
     // * or it's not in localStorage (it hasn't been dismissed yet)
-    if (isDismissable && (!notificationId || !localStorage.getItem(notificationId))) {
+    if (!notificationId || (isDismissable && !localStorage.getItem(notificationId))) {
```

Comment:
I _think_ this was right as-is? But it's been a while since I've taken a look at this, could you go into a bit more detail why you think this should be changed?

Also, if it does need changing, we should update the comment as well.

Reply:
the reason is that ephemeral notifications should always be visible, regardless of the dismissability of the notifications.

The logic I'm trying to follow, and is not fully covered by code and tests atm is to show the notification if:

- the notification is ephemeral, i.e. it has no notification ID
- it's not ephemeral and is not dismissable
- it's not ephemeral, is dismissable and the user has not dismissed it yet

If that makes sense I can add the necessary test cases to cover all those scenarios

Reply:
OK, that makes sense, and I can see the edge case that this change is covering. I don't think additional tests are needed but it would be good to update the comment.